File: bottomsheetmenu/src/main/java/com/strekha/bottomsheetmenu/Item.java
package com.strekha.bottomsheetmenu;

import android.support.annotation.NonNull;
import android.support.v4.internal.view.SupportMenuItem;
import android.view.MenuItem;

abstract class Item {

    abstract int getWeight(int maxWeight);

    static class Title extends Item {

        @NonNull
        final CharSequence title;

        Title(@NonNull CharSequence title) {
            this.title = title;
        }

        @Override
        int getWeight(int maxWeight) {
            return maxWeight;
        }
    }

    static class Element extends Item {

        @NonNull
        final SupportMenuItem item;

        Element(@NonNull SupportMenuItem item) {
            this.item = item;
        }

        @Override
        int getWeight(int maxWeight) {
            return 1;
        }
    }

    static class Separator extends Item {

        @Override
        int getWeight(int maxWeight) {
            return maxWeight;
        }
    }
}

File: bottomsheetmenu/src/main/java/com/strekha/bottomsheetmenu/Item.kt
package com.strekha.bottomsheetmenu

import android.support.v4.internal.view.SupportMenuItem

internal sealed class Item {

    abstract fun getWeight(maxWeight: Int): Int

    class Title(val title: CharSequence) : Item() {

        override fun getWeight(maxWeight: Int): Int {
            return maxWeight
        }
    }

    class Element(val item: SupportMenuItem) : Item() {

        override fun getWeight(maxWeight: Int): Int {
            return 1
        }
    }

    object Separator : Item() {

        override fun getWeight(maxWeight: Int): Int {
            return maxWeight
        }
    }
}
